File: backend-python/models/appointment.py
from sqlalchemy import Column, String, Date, DateTime, ForeignKey, Text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from database.database import Base
import uuid

class Appointment(Base):
    __tablename__ = "appointments"
    
    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    user_id = Column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=False)
    doctor_name = Column(String(255), nullable=False)
    specialization = Column(String(255), nullable=False)
    date = Column(Date, nullable=False)
    time = Column(String(50), nullable=False)
    status = Column(String(50), default="Pending")  # Pending, Confirmed, Completed
    notes = Column(Text, nullable=True)
    created_at = Column(DateTime, nullable=False)
    
    # Relationship
    user = relationship("User", back_populates="appointments")

File: backend/src/entities/Appointment.ts
import "reflect-metadata";
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, ManyToOne, JoinColumn } from "typeorm";
import { User } from "./User";

@Entity("appointments")
export class Appointment {
  @PrimaryGeneratedColumn("uuid")
  id: string;

  @Column({ type: "uuid" })
  userId: string;

  @ManyToOne(() => User, user => user.appointments)
  @JoinColumn({ name: "userId" })
  user: User;

  @Column({ type: "varchar", length: 255 })
  doctorName: string;

  @Column({ type: "varchar", length: 255 })
  specialization: string;

  @Column({ type: "date" })
  date: Date;

  @Column({ type: "varchar", length: 50 })
  time: string;

  @Column({ type: "varchar", length: 50, default: "Pending" })
  status: string; // Pending, Confirmed, Completed

  @Column({ type: "text", nullable: true })
  notes: string;

  @CreateDateColumn()
  createdAt: Date;
}